nomad/fsm.go

```diff
@@ -204,6 +207,49 @@ func (n *nomadFSM) applyUpsertJob(buf []byte, index uint64) interface{} {
 		n.logger.Printf("[ERR] nomad.fsm: UpsertJob failed: %v", err)
 		return err
 	}
+
+	// If it is periodic, insert it into the periodic runner and record the
+	// time it was inserted.
+	if req.Job.IsPeriodic() {
+		if err := n.periodicRunner.Add(req.Job); err != nil {
+			n.logger.Printf("[ERR] nomad.fsm: PeriodicRunner.Add failed: %v", err)
+			return err
+		}
+
+		// Record the insertion time as a launch.
```

Comment:
Can you add more comments explaining why this is necessary. I get it, but its not immediately obvious.